forms/models/elements/subform.js

```diff
@@ -256,7 +256,14 @@ define(function (require) {
 
       return new Promise(function (resolve, reject) {
         Promise.all(promises).then(function (values) {
-          resolve(values);
+          values = _.reduce(values, function (memo, value) {
+            if (value._action === 'remove' && (value.id === null || value.id === undefined)) {
+              return memo;
+            }
+            memo.push(value);
+            return memo;
+          }, []);
+          resolve(_.compact(values));
```

Comment:
I'm curious about the `_.compact()` here. It only removes falsey values, which means empty Objects will still be present afterwards.

For that reason, I think it'd probably be faster (1 less loop) and simpler to just use either `_.without()` or `Array#filter()` instead of both `_.reduce()` and `_.compact`. You'll be able to exclude more aggressively in a combined filter function.